internal/workflow/activity/latest_block.go

```diff
@@ -0,0 +1,77 @@
+package activity
+
+import (
+	"context"
+	"github.com/coinbase/chainstorage/internal/cadence"
+	"github.com/coinbase/chainstorage/internal/config"
+	"github.com/coinbase/chainstorage/internal/gateway"
+	"github.com/coinbase/chainstorage/internal/utils/fxparams"
+	api "github.com/coinbase/chainstorage/protos/coinbase/chainstorage"
+	"go.temporal.io/sdk/workflow"
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+	"golang.org/x/xerrors"
+)
+
+type (
+	LatestBlock struct {
+		baseActivity
+		config *config.Config
+		logger *zap.Logger
+		client gateway.Client
+	}
+
+	LatestBlockParams struct {
+		fx.In
+		fxparams.Params
+		Runtime cadence.Runtime
+		Client  gateway.Client
+	}
+
+	LatestBlockRequest struct {
```

Comment:
nit: instead of reading the IrreversibleDistance from cfg, pass it in from the request

Reply:
Fixed.